cluster/certificates.go

```diff
@@ -176,3 +181,28 @@ func saveCertToKubernetes(kubeClient *kubernetes.Clientset, crtName string, crt
 		return fmt.Errorf("[certificates] Timeout waiting for kubernetes to be ready")
 	}
 }
+
+func deployBackupCertificates(ctx context.Context, backupHosts []*hosts.Host, kubeCluster *Cluster) error {
+	var errgrp errgroup.Group
+
+	for _, host := range backupHosts {
+		runHost := host
+		errgrp.Go(func() error {
+			return pki.DeployCertificatesOnHost(ctx, runHost, kubeCluster.Certificates, kubeCluster.SystemImages.CertDownloader, pki.TempCertPath, kubeCluster.PrivateRegistriesMap)
+		})
+	}
+	return errgrp.Wait()
+}
+
+func fetchBackupCertificates(ctx context.Context, backupHosts []*hosts.Host, kubeCluster *Cluster) (map[string]pki.CertificatePKI, error) {
+	for _, host := range backupHosts {
+		certificates, err := pki.FetchCertificatesFromHost(ctx, kubeCluster.EtcdHosts, host, kubeCluster.SystemImages.Alpine, kubeCluster.LocalKubeConfigPath, kubeCluster.PrivateRegistriesMap)
```

Comment:
Only one host has to succeed, it seems like if the first fails, it will give up and not try the second.  Is that true?

Reply:
Not exactly. `pki.FetchCertificatesFromHost()` handles file not found internally, so `err != nil` will an actual error: something happened while doing a docker command, certificate parsing error..etc.